src/components/localisation/routes.ts

```diff
@@ -1,13 +1,181 @@
 import path from 'path'
-import { getAppEntryPointDir } from '../../components/utilityFunctions'
+import { rmdirSync, readFileSync, writeFile } from 'fs'
+import { promisify } from 'util'
+import {
+  getAppEntryPointDir,
+  combineRequestParams,
+  makeFolder,
+} from '../../components/utilityFunctions'
 import config from '../../config'
 
 const { localisationsFolder } = config
 
+const writeFilePromise = promisify(writeFile)
+
+export type LanguageOption = {
+  languageName: string
+  description: string
+  code: string
+  flag: string // To-do: limit to flag emojis
+  enabled: boolean
+}
+
 export const routeGetLanguageFile = async (request: any, reply: any) => {
   const { code } = request.params
+  const languageOptions = readLanguageOptions()
+  const requestedIndex = languageOptions.findIndex((lang: LanguageOption) => code === lang.code)
+
+  if (requestedIndex === -1) {
+    reply.statusCode = 400
+    return reply.send({
+      error: 'Language error',
+      message: 'Language code not recognised',
+    })
+  }
+
+  if (!languageOptions[requestedIndex].enabled) {
+    reply.statusCode = 403
+    return reply.send({
+      error: 'Language error',
+      message: 'Language code not enabled',
+    })
+  }
+
   const stringsFile = path.join(code, 'strings.json')
   reply.sendFile(stringsFile, path.join(getAppEntryPointDir(), localisationsFolder))
 }
 
-export const updateLanguageFiles = async (request: any, reply: any) => {}
+export const routeGetAllLanguageFiles = async (request: any, reply: any) => {
+  const languageOptions = readLanguageOptions()
+  const output: { [key: string]: { [key: string]: string } } = {}
+
+  for (const { code } of languageOptions) {
+    const stringsFile = path.join(code, 'strings.json')
+    output[code] = JSON.parse(
+      readFileSync(path.join(getAppEntryPointDir(), localisationsFolder, stringsFile), 'utf-8')
+    )
+  }
+  reply.send(output)
+}
+
+export const routeEnableLanguage = async (request: any, reply: any) => {
+  const { code, enabled } = combineRequestParams(request)
+  const languageOptions: LanguageOption[] = readLanguageOptions()
+  const index = languageOptions.findIndex((language) => language.code === code)
+  if (index === -1) return reply.send({ success: false, message: 'Code not recognised' })
+
+  const changedLanguage = {
+    ...languageOptions[index],
+    enabled:
+      enabled === 'true' ? true : enabled === 'false' ? false : !languageOptions[index].enabled,
+  }
+  languageOptions[index] = changedLanguage
+
+  if (languageOptions.filter((language) => language.enabled).length === 0)
+    return reply.send({ success: false, message: 'At least one language must remain enabled' })
```

Comment:
Do we need to localise these strings? I'm thinking about a local Admin person using another default language

Reply:
I've been avoiding trying to localise server strings, cos it seems like a bit of a headache. We'll need a general solution at some point.

My thinking here was that anyone doing language management would need to be familiar with English anyway, since all the default strings are in English and you can't really make sense of the exported CSV without knowing English.

Reply:
But the front-end ones are localised anyway

Reply:
That's a good point. I didn't realise it was back-end, far point.